cmd/registry/patch/artifact.go

```diff
@@ -196,27 +169,6 @@ func applyArtifactPatch(ctx context.Context, client connection.RegistryClient, c
 	return nil
 }
 
-// populateIdAndKind inserts the "id" and "kind" fields in the supplied json bytes.
-func populateIdAndKind(bytes []byte, kind, id string) ([]byte, error) {
```

Comment:
I think this will break API hub, which currently reads proto-serialized artifacts and expects them to contain "kind" and "id" fields. Since these fields are redundant in Registry YAML, I don't think we should be including them in YAML-serialized artifacts and I expect that API hub will adapt to that when it starts reading those YAML-serialized artifacts.

Reply:
That's unfortunate. So I need to add this back for protos?

Reply:
Actually, this PR isn't critical until we go all in on yaml...perhaps we should hold it until we correct API Hub.